minion/server/routes.py

```diff
@@ -43,6 +45,7 @@ def get_status():
     return jsonify(status)
 
 
+@ensure_mongo
```

Comment:
Here you do not want to decorate the `@api.route()` decorator but the `get_jobs()` function - Thus, change the order of decorators:

```python
@api.route(...)
@ensure_mongo
def get_jobs():
   ...
```

Internally Python will do something like this (simplified):

```python
api.route(ensure_mongo(get_jobs))
```

So the `api.route` actually gets the `ensure_mongo`'s return value, which is `func_wrapper` (your decorator wrapper).

If you do it the other way around it would look like this:

```python
ensure_mongo(api.route(get_jobs))
```

In this scenario `api.route` won't even know about the `func_wrapper` returned by `ensure_mongo` because it gets the `get_jobs` function as argument.